GDJS/GDJS/IDE/UsedResourcesDeclarer.h

```diff
@@ -0,0 +1,86 @@
+/*
+ * GDevelop JS Platform
+ * Copyright 2008-2023 Florian Rival (Florian.Rival@gmail.com). All rights
+ * reserved. This project is released under the MIT License.
+ */
+
+#pragma once
+
+#include "GDCore/IDE/Project/ArbitraryResourceWorker.h"
+#include "GDCore/String.h"
+
+#include <set>
+
+namespace gd {
+class Project;
+class Layout;
+class SerializerElement;
+} // namespace gd
+
+namespace gd {
+
+/**
+ * \brief Find resource usages in several parts of the project.
+ *
+ * \ingroup IDE
+ */
+class UsedResourcesFinder : private gd::ArbitraryResourceWorker {
```

Comment:
We can't use use `ResourcesInUseHelper`?

Reply:
Do you mean moving the features in this class?

Reply:
`ResourcesInUseHelper` search resources by type on the whole project when the new one is searching resources per scene no matter the type. It's very likely that we will want to cross these searches at some point, but we should leave it for later. 
I should rename the new class `SceneResourcesFinder` and move it in the same folder as `ResourcesInUseHelper`.